opal-ui/src/components/admin/providers/AddIdentityProviderDialog.vue

```diff
@@ -0,0 +1,348 @@
+  name: 'AddIdentityProviderDialog',
+});
+</script>
+<script setup lang="ts">
+import { IDProviderDto } from 'src/models/Opal';
+import { notifyError } from 'src/utils/notify';
+
+interface DialogProps {
+  modelValue: boolean;
+  provider: IDProviderDto | null;
+}
+
+const { t } = useI18n();
+const identityProvidersStore = useIdentityProvidersStore();
+const formRef = ref();
+const props = defineProps<DialogProps>();
+const emit = defineEmits(['update:modelValue']);
+const showDialog = ref(props.modelValue);
+const emptyProvider = {
+  name: '',
+  clientId: '',
+  secret: '',
+  discoveryURI: '',
+  scope: 'openid',
+  useNonce: true,
+  useLogout: true,
+  parameters: [],
+  enabled: false,
+  connectTimeout: 0,
+  readTimeout: 0,
+} as IDProviderDto;
+
+const groupsMappingOptionsIndices = {
+  groupsClaim: 0,
+  groupsScript: 1,
+};
+const groupsMappingOptions = ref([
+  { label: t('identity_provider.groups_claim'), value: 'groupsClaim' },
+  { label: t('identity_provider.groups_javascript'), value: 'groupsScript' },
+]);
+const groupsMapping = ref(groupsMappingOptions.value[groupsMappingOptionsIndices.groupsClaim]);
+const newProvider = ref<IDProviderDto>(emptyProvider);
+const editMode = computed(() => !!props.provider && !!props.provider.name);
+const submitCaption = computed(() => (editMode.value ? t('update') : t('add')));
+const dialogTitle = computed(() => (editMode.value ? t('identity_provider_edit') : t('identity_provider_add')));
+const isGroupsMappingByClaim = computed(() => groupsMapping.value.value === 'groupsClaim');
+const groupsJavascriptPlaceholder = computed(() => {
+  return `// input: userInfo
+// output: an array of strings
+
+// example:
+userInfo.some.property.map(x => x.split(':')[0])
+`;
+});
+const discoveryURIDefinition = computed(
+  () =>
+    `<a href="https://openid.net/specs/openid-connect-discovery-1_0.html" target="_blank" tabindex="-1">${t(
+      'identity_provider.discovery_uri_definition'
+    )}</a>`
+);
+const useNonceDefinition = computed(
+  () =>
+    `<a href="https://openid.net/specs/openid-connect-core-1_0.html#IDToken" target="_blank" tabindex="-1">${t(
+      'identity_provider.use_nonce_definition'
+    )}</a>`
+);
+const useLogoutDefinition = computed(
+  () =>
+    `<a href="https://openid.net/specs/openid-connect-session-1_0-17.html#RPLogout" target="_blank" tabindex="-1">${t(
+      'identity_provider.use_logout_definition'
+    )}</a>`
+);
+
+// Validation rules
+const validateRequiredField = (id: string) => (val: string) => (val && val.trim().length > 0) || t(id);
+const validateUri = (id: string, required: boolean) => (val: string) => {
+  if (!val && !required) {
+    return true;
+  }
+
+  return new RegExp(/^(http|https):\/\/[^ "]+$/).test(val) || t(id);
+};
+
+watch(
+  () => props.modelValue,
+  (value) => {
+    if (value) {
+      if (props.provider) {
+        newProvider.value = { ...props.provider };
+        groupsMapping.value = newProvider.value.groupsClaim
```

Comment:
When groupsClaim is "", the wrong groups mapping strategy is selected, you must check for undefined value instead

Reply:
:+1: